File: warmup/sherlockgcd/sherlockgcd.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"strconv"
	"strings"
)

func main() {
	//file := os.Stdin
	file, _ := os.Open(`.\data\sherlockgcd.txt`)
	rawBytes, _ := ioutil.ReadAll(file)
	cases := strings.Fields(string(rawBytes))[1:]

	// For each test case..
	for i := 0; i < len(cases); {
		length, _ := strconv.Atoi(cases[i])
		div, ans := 0, "NO"

		// Parse numbers
		for j := 0; j < length; j++ {
			n, _ := strconv.Atoi(cases[i+j+1])

			if div == 0 {
				div = n
			} else {
				div = gcd(div, n)
			}

			if div == 1 {
				ans = "YES"
				break
			}
		}

		// Print the answer for this case
		fmt.Println(ans)

		// Continue to next test case set
		i += length + 1
	}
}

func gcd(x, y int) int {
	for y != 0 {
		x, y = y, x%y
	}
	return x
}

File: warmup/sherlockgcd/sherlockgcd.py
#from sys import stdin

data = open( r".\data\sherlockgcd.txt" )

def gcd(a, b):
    if b == 0:
        return a
    else:
        return gcd(b, a % b)

numCases = int(data.readline())

for _ in range(numCases):
	data.readline() # skip N
	nums = list(map(int, data.readline().strip().split()))

	g = nums[0]
	for n in nums:
		g = gcd(g, n)

	if g == 1:
		print("YES")
	else:
		print("NO")

data.close()
